dom-fetch-client/src/main/scala/org/http4s/dom/FetchClientBuilder.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.dom
+
+import cats.effect.Async
+import cats.effect.Resource
+import org.http4s.client.{Client, defaults}
+import org.http4s.internal.BackendBuilder
+import org.scalajs.dom.experimental.ReferrerPolicy
+import org.scalajs.dom.experimental.RequestCache
+import org.scalajs.dom.experimental.RequestCredentials
+import org.scalajs.dom.experimental.RequestMode
+import org.scalajs.dom.experimental.RequestRedirect
+import scala.concurrent.duration._
+
+/** Configure and obtain a FetchClient
+  *
+  * @param requestTimeout maximum duration from the submission of a request through reading the body before a timeout.
+  * @param cache how the request will interact with the browser’s HTTP cache
+  * @param credentials what browsers do with credentials (cookies, HTTP authentication entries, and TLS client certificates)
+  * @param mode mode you want to use for the request, e.g., cors, no-cors, or same-origin
+  * @param redirect how to handle a redirect response
+  * @param referrer referrer of the request
+  * @param referrerPolicy referrer policy to use for the request
+  */
+sealed abstract class FetchClientBuilder[F[_]] private (
+    val requestTimeout: Duration,
+    val cache: Option[RequestCache],
+    val credentials: Option[RequestCredentials],
+    val mode: Option[RequestMode],
+    val redirect: Option[RequestRedirect],
+    val referrer: Option[FetchReferrer],
+    val referrerPolicy: Option[ReferrerPolicy]
+)(implicit protected val F: Async[F])
```

Comment:
Can you add an `override` here since IIRC we get this from the `BackendBuilder`?